test/csuite/timestamp_abort/main.c

```diff
@@ -340,6 +371,241 @@ set_flush_tier_delay(WT_RAND_STATE *rnd)
+
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", WT_HOME_DIR, filename));
+                    testutil_assert((rfd = open(buf, O_RDONLY, 0666)) >= 0);
+
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", backup_home, filename));
+                    testutil_assert((wfd = open(buf, O_WRONLY | O_CREAT, 0666)) >= 0);
+                }
+
+                /* Copy the range. */
+                tmp = dcalloc(1, size);
+                rdsize = pread(rfd, tmp, (size_t)size, (wt_off_t)offset);
+                testutil_assert(rdsize >= 0);
+                testutil_assert(pwrite(wfd, tmp, (size_t)rdsize, (wt_off_t)offset) == rdsize);
+                free(tmp);
+
+                nranges++;
+            } else {
+
+                /* We are supposed to do the full file copy. */
+                testutil_assert(first_range);
+                first_range = false;
+                testutil_system("cp %s/%s %s/%s", WT_HOME_DIR, filename, backup_home, filename);
+            }
+        }
+
+        if (rfd > 0)
+            testutil_assert(close(rfd) == 0);
+        if (wfd > 0)
+            testutil_assert(close(wfd) == 0);
+
+        if (first_range) {
+            /* "Copy" the file from the source backup (actually, just do a hard link!). */
+            testutil_system("ln %s/%s %s/%s", src_backup_home, filename, backup_home, filename);
+            nunmodified++;
+        }
+
+        testutil_assert(ret == WT_NOTFOUND);
+        testutil_check(file_cursor->close(file_cursor));
+    }
+    testutil_assert(ret == WT_NOTFOUND);
+
+    /* Cleanup */
+    testutil_check(cursor->close(cursor));
+    testutil_check(session->close(session, NULL));
+
+    /* Remember that the backup finished successfully. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/done", backup_home));
+    testutil_assert((wfd = open(buf, O_WRONLY | O_CREAT, 0666)) >= 0);
+    testutil_assert(close(wfd) == 0);
+
+    printf("Create backup: Files: %" PRId32 ", Ranges: %" PRId32 ", Unmodified: %" PRId32 "\n",
+      nfiles, nranges, nunmodified);
+
+    /* Immediately verify the backup. */
+    if (backup_verify_throughout) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "ID%" PRIu32, index));
+        testutil_verify_src_backup(conn, backup_home, WT_HOME_DIR, buf);
+    }
+}
+
+/*
+ * backup_delete_old_backups --
+ *     Delete old backups, keeping just a few recent ones, so that we don't take too much space for
+ *     no good reason.
+ */
+static void
+backup_delete_old_backups(int retain)
+{
+    struct dirent *dir;
+    struct stat sb;
+    DIR *d;
+    int count, i, indexes[256];
+    char buf[256];
+    bool done;
+
+    do {
+        done = true;
+        testutil_assert_errno((d = opendir(".")) != NULL);
+        count = 0;
+        while ((dir = readdir(d)) != NULL) {
+            if (strncmp(dir->d_name, "backup.", 7) == 0) {
+                indexes[count++] = atoi(dir->d_name + 7);
```

Comment:
Minor nit: I suggest making "backup." a global macro since it is used elsewhere and then here:
len = strlen(BACKUP_BASE); instead of "7" in both lines.